### DISCUSSION TITLE
why the WriteableBitmap doesn't work?

### DISCUSSION THREAD

**Discussion by myot233** [microsoft/microsoft-ui-xaml#9466] [Q&A] [answered]
I'd like to call the method shown in the image below to do some image-related processing, but I've found that if I add an arbitrary asynchronous statement above the  ```var data = new byte[415800];``` the WriteableBitmap returned by this function won't be displayed, and I'm also trying to read out the pixelbuffer, which is exactly what I entered.
![image](https://github.com/microsoft/microsoft-ui-xaml/assets/66019007/6a6302b6-1239-423c-8556-a2f592397614)

this is where the method is called
![image](https://github.com/microsoft/microsoft-ui-xaml/assets/66019007/633c82ca-48f5-4633-91fc-0c7c8e5ed6d5)

if there are more easy way to do some image-related processing in WinUI3,please tell me about it

New to WinUI3 and C# Asynchronous programming, I don't understand why it doesn't work。

**Comment by lhak** [comment]
How does your XAML code look like? I guess you also need to call Invalidate() on AnimImageSource and not img.

**Comment by castorix** [comment, marked as answer]
> if there are more easy way to do some image-related processing in WinUI3,please tell me about it

MS samples use Win2D and [GaussianBlurEffect Class](https://microsoft.github.io/Win2D/WinUI2/html/T_Microsoft_Graphics_Canvas_Effects_GaussianBlurEffect.htm) for Blur effect, like in [WindowsCompositionSamples](https://github.com/microsoft/WindowsCompositionSamples)
I prefer to use directly Direct2D and [Gaussian blur effect](https://learn.microsoft.com/en-us/windows/win32/direct2d/gaussian-blur)
A test with D2D1_GAUSSIANBLUR_PROP_STANDARD_DEVIATION  = 3 and 10 :

![orig](https://github.com/microsoft/microsoft-ui-xaml/assets/22345506/59e1a00f-747f-4897-9f75-c26257982fea)

![result_blur3](https://github.com/microsoft/microsoft-ui-xaml/assets/22345506/b14e976b-8514-455b-a39e-fcd96b0dfd80)
![result_blur_10](https://github.com/microsoft/microsoft-ui-xaml/assets/22345506/e7e6ffd6-bf35-4863-b3f8-4eeda3079c37)

**Comment by castorix** [comment]
I just uploaded a test sample with most of Direct2D effects : [WinUI3_Direct2D_Effects](https://github.com/castorix/WinUI3_Direct2D_Effects/)
I added some image transiton animations (I tried to find ideas from Google, mainly animations I found in Adobe After Effects demos, but some are difficult to reproduce...)
